grammar.js

```diff
@@ -1163,7 +1169,7 @@ module.exports = grammar(C, {
         alias($.dependent_identifier, $.dependent_name),
         $.qualified_identifier,
         $.template_function,
-        $.identifier,
+        seq(optional('template'), $.identifier),
```

Comment:
This is a band aid for the actual issue and isn't generally scalable.

The issue is that it should have matched a qualified _type_ identifier which would properly match the template keyword before the typename.

I haven't played with it yet but either making a qualified type identifier a valid child of normal qualified identifier, or rewriting the rules not not distinguish between them, or at least allowing either rule to match would be better than this I think.